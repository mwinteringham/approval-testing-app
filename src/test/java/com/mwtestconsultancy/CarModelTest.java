package com.mwtestconsultancy;

import org.approvaltests.Approvals;
import org.junit.Test;

import java.time.LocalDate;

public class CarModelTest {

    @Test
    public void testModel(){
        CarModel carModel = new CarModel("SEAT", "Ibiza", 4000, LocalDate.of(2018, 01, 01), "/link/to/car");

        Approvals.verify(carModel);
    }

    @Test
    public void testInitialToggleState(){
        CarDatabase carDatabase = new CarDatabase();
        CarModel carOne = carDatabase.getCurrentCar();

        Approvals.verify(carOne.toString());
    }

    @Test
    public void testTogglingState(){
        CarDatabase carDatabase = new CarDatabase();
        carDatabase.switchCars();
        CarModel carTwo = carDatabase.getCurrentCar();

        Approvals.verify(carTwo);
    }

}
