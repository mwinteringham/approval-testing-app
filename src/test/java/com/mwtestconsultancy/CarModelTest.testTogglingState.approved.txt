CarModel{company='Toyota', make='Auris', price=28000, datePurchased=2017-03-21, image='./auris.jpg'}
